management/models.py

```diff
@@ -9,3 +9,45 @@ class User(AbstractUser):
     """
 
     pass
+
+
+class PermissionsBase(models.Model):
+    """Base model for models that require permissions."""
+
+    owner = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
+
+    PERMISSIONS_LEVELS = [
+        ("public", "Public"),
+        ("private", "Private"),
+    ]
+
+    permissions_level = models.CharField(
+        max_length=8, choices=PERMISSIONS_LEVELS, default="private"
+    )
+
+    def set_permissions(self):
+        """Set object-level permissions."""
+
+        delete, change, view = _get_perm_codenames(self.__class__)
+
+        # Assign view permissions for all users
+        assign_perm(view, User.objects.all(), self)
```

Comment:
All users should have `view` access if the permission level is `Public` but not if `Private`, right?

Reply:
And I guess we need to consider what happens with the `internal` case. Possibly via groups, as you mentioned.